File: .cpp/287.FindTheDuplicateNumber.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int n = nums.size();
        int i = 0;

        while (i < n)
        {
            int value = nums[i];
            if (value > 0 && value <= n && nums[value - 1] != value)
            {
                swap(nums[i], nums[value - 1]);
            }
            else
            {
                i++;
            }
        }
        return nums[n-1];
    }
};

File: 287.FindTheDuplicateNumber.py
class Solution:
    def findDuplicate(self, nums: List[int]) -> int:
        n = len(nums)
        i = 0
        while i<n:
            value = nums[i] - 1

            check = value >= 0 and value < n

            if check and nums[i]!=nums[value]:
                nums[i],nums[value] = nums[value],nums[i]
            else:
                i+=1
        return nums[n-1]